source/diffHandler.py

```diff
@@ -0,0 +1,189 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2020 Bill Dengler
+
+import config
+import globalVars
+import os
+import struct
+import subprocess
+import sys
+from abc import abstractmethod
+from baseObject import AutoPropertyObject
+from difflib import ndiff
+from logHandler import log
+from textInfos import TextInfo, UNIT_LINE
+from threading import Lock
+from typing import List
+
+
+class DiffAlgo(AutoPropertyObject):
+	@abstractmethod
+	def diff(self, newText: str, oldText: str) -> List[str]:
+		raise NotImplementedError
+
+	@abstractmethod
+	def _getText(self, ti: TextInfo) -> str:
+		raise NotImplementedError
+
+
+class DiffMatchPatch(DiffAlgo):
+	"""A character-based diffing approach, using the Google Diff Match Patch
+	library in a proxy process (to work around a licence conflict).
+	"""
+	#: A subprocess.Popen object for the nvda_dmp process.
+	_proc = None
+	#: A lock to control access to the nvda_dmp process.
+	#: Control access to avoid synchronization problems if multiple threads
+	#: attempt to use nvda_dmp at the same time.
+	_lock = Lock()
+
+	def _initialize(self):
+		"Start the nvda_dmp process if it is not already running."
+		if not DiffMatchPatch._proc:
+			log.debug("Starting diff-match-patch proxy")
+			if hasattr(sys, "frozen"):
+				dmp_path = (os.path.join(globalVars.appDir, "nvda_dmp.exe"),)
+			else:
+				dmp_path = (sys.executable, os.path.join(
+					globalVars.appDir, "..", "include", "nvda_dmp", "nvda_dmp.py"
+				))
+			DiffMatchPatch._proc = subprocess.Popen(
+				dmp_path,
+				creationflags=subprocess.CREATE_NO_WINDOW,
+				bufsize=0,
+				stdin=subprocess.PIPE,
+				stdout=subprocess.PIPE
+			)

```

Comment:
No, because:

* client code calls `diff`, which needs to acquire the lock.
* If nvda_dmp is not running, `diff` calls `_initialize`.
* with your changes, `_initialize` waits to acquire the lock (held by `diff`).
* Deadlock.

Note that `diff` is the only function that calls `_initialize`.